subsys/mpsl/pm/mpsl_pm_utils.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <mpsl_pm.h>
+#include <mpsl_pm_config.h>
+#include <zephyr/pm/policy.h>
+#include <zephyr/logging/log.h>
+
+#include <mpsl/mpsl_work.h>
+#include <mpsl/mpsl_pm_utils.h>
+
+LOG_MODULE_REGISTER(mpsl_pm_utils, CONFIG_MPSL_LOG_LEVEL);
+
+/* These values are still arbritrary*/
```

Comment:
As we cannot predict what gets executed in between in high prio, nor really how many non-PM items are in the workqueue beforehand, this is very hard to determine. Testing it is currently also not viable, as the zephyr implementation is non functional. (will be worked on soon--I hope)